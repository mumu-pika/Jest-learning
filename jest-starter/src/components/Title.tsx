import React, { CSSProperties, FC } from 'react'

interface Props {
  type: 'large' | 'small'
  title: string
}

// large 样式
const largeStyle: CSSProperties = {
  fontSize: '2em',
  color: 'red',
}

// small 样式
const smallStyle: CSSProperties = {
  fontSize: '1em',
  color: 'green',
}

// 样式 Mapper
const styleMapper: Record<'small' | 'large', CSSProperties> = {
  small: smallStyle,
  large: largeStyle,
}

// 组件
const Title: FC<Props> = (props) => {
  const { title, type } = props

  return <p style={styleMapper[type]}>{title}</p>
}

export default Title
