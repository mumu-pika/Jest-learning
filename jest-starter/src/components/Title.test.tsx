// tests/components/Title.test.tsx
// 快照测试
import React from "react";
import { render } from "@testing-library/react";
import Title from "components/Title";

describe("Title", () => {
  it("可以正确渲染大字", () => {
    const { baseElement } = render(<Title type="large" title="大字" />);
    expect(baseElement).toMatchSnapshot();
  });

  it("可以正确渲染小字", () => {
    const { baseElement } = render(<Title type="small" title="小字" />);
    expect(baseElement).toMatchSnapshot();
  });
});
