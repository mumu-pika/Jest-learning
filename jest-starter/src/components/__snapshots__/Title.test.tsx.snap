// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title 可以正确渲染大字 1`] = `
<body>
  <div>
    <p
      style="font-size: 2em; color: red;"
    >
      大字
    </p>
  </div>
</body>
`;

exports[`Title 可以正确渲染小字 1`] = `
<body>
  <div>
    <p
      style="font-size: 1em; color: green;"
    >
      小字
    </p>
  </div>
</body>
`;
